spec/controllers/idv_controller_spec.rb

```diff
@@ -55,6 +55,21 @@
       expect(response).to redirect_to idv_doc_auth_path
     end
 
+    it 'redirects to inherited proofing if an auth code parameter is included in request URL' do
+      session_service_provider = create(:service_provider)
+      users_auth_code = 'any_auth_code'
+      request_url = "https://login.gov/openid_connect/authorize?inherited_proofing_auth=#{users_auth_code}"
+
+      allow_any_instance_of(ApplicationController).to receive(:current_sp).
+        and_return(session_service_provider)
+      stub_sign_in
+      session[:sp] = { request_url: request_url }
+
+      get :index
+
+      expect(response).to redirect_to idv_inherited_proofing_path
+    end
```

Comment:
This may come down to personal testing philosophies, but to me this seems pretty tightly coupled with the specific, current implementation of `va_inherited_proofing?`, where I might leave those considerations to the specs of the concern itself, and instead stub the expected return values here.

```suggestion
    context 'with a va inherited proofing session' do
      before do
        stub_sign_in
        allow(controller).to receive(:va_inherited_proofing?).and_return(true)
      end

      it 'redirects to inherited proofing' do
        get :index

        expect(response).to redirect_to idv_inherited_proofing_path
      end
    end
```

Reply:
Andrew, thanks for the sage advice. Yes, I agree (after you pointed it out), the test shouldn't be coupled to the concerns logic; but I did need to see (temporarily) that ApplicationController had access to the DecoratedSession, that the InheritedProofingConcern depends upon; I'll use your version of the test. Now, to backfill the missing concern tests. :-/

Reply:
Regarding use of the _auth_code within the request_url_, I may have an issue with how we've conflated _inherited proofing_ with _VA inherited proofing_. What will happen when we next have a new service provider XYZ that sends an auth_code, will we direct them to the VA inherited proofing?  Seems we should be using the ServiceProvider object. We can cross that bridge when we get to the bridge. :-)